frappe/core/doctype/file/file.py

```diff
@@ -466,10 +467,16 @@ def save_file(self, content=None, decode=False, ignore_existing_file_check=False
 
 		if not self.is_private:
 			self.is_private = 0
-		self.file_size = self.check_max_file_size()
-		self.content_hash = get_content_hash(self.content)
+		
 		self.content_type = mimetypes.guess_type(self.file_name)[0]
+		
+		if self.content_type and "image" in self.content_type and \
+			frappe.get_system_settings("remove_exif_tags"):
+			self.content = strip_exif_data(self.content, self.content_type)			
 
+		self.file_size = self.check_max_file_size()
+		self.content_hash = get_content_hash(self.content)
```

Comment:
```suggestion
		self.file_size = self.check_max_file_size()
		if (
			self.content_type and "image" in self.content_type 
			and frappe.get_system_settings("remove_exif_tags")
		):
			self.content = strip_exif_data(self.content, self.content_type)			

		self.content_hash = get_content_hash(self.content)
```

1. This way of formatting conditions seems cleaner IMO
2. Check for max file size prior to EXIF stripping might save us one potentially unnecessary step.

Reply:
Makes sense, will change.

Reply:
will do